GroceryApplication/app/src/main/java/com/company/arodrigues/groceryapplication/MainActivityFragment.java

```diff
@@ -0,0 +1,22 @@
+package com.company.arodrigues.groceryapplication;
+
+import android.support.v4.app.Fragment;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+/**
+ * A placeholder fragment containing a simple view.
+ */
+public class MainActivityFragment extends Fragment {
+
+    public MainActivityFragment() {
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
```

Comment:
So for this lab we weren't expecting you to use  a fragment. However, it seems that you have added the actual xml for everything that displays to the fragment's xml file file setting all the onclick listeners in the OnCreate of the activity. While functionally this works it is not best practice. You should define listeners for a particular view in that view's java file.